readme.md

```diff
@@ -840,19 +840,51 @@ try {
 
 ### Execute the current package's binary
 
+Execa can be combined with [`get-bin-path`](https://github.com/ehmicky/get-bin-path) to test the current package's binary. As opposed to hard-coding the path to the binary, this validates that the `package.json` `bin` field is correctly set up.
+
 ```js
 import {getBinPath} from 'get-bin-path';
 
 const binPath = await getBinPath();
 await execa(binPath);
 ```
 
-`execa` can be combined with [`get-bin-path`](https://github.com/ehmicky/get-bin-path) to test the current package's binary. As opposed to hard-coding the path to the binary, this validates that the `package.json` `bin` field is correctly set up.
+### Ensuring `all` output is interleaved
+
+The `all` [stream](#all) and [string/`Uint8Array`](#all-1) properties are guaranteed to interleave [`stdout`](#stdout) and [`stderr`](#stderr).
+
+However, for performance reasons, the child process might buffer and merge multiple simultaneous writes to `stdout` or `stderr`. This prevents proper interleaving.
+
+For example, this prints `1 3 2` instead of `1 2 3` because both `console.log()` are merged into a single write.
+
+```js
+import {execa} from 'execa';
+
+const {all} = await execa('node', ['example.js'], {all: true});
+console.log(all);
+```
+
+```js
+// example.js
+console.log('1'); // writes to stdout
+console.error('2'); // writes to stderr
+console.log('3'); // writes to stdout
+```
+
+This can be worked around by using `process.nextTick()` or `setTimeout()`.
+
+```js
+console.log('1');
+console.error('2');
+process.nextTick(() => {
```

Comment:
[queueMicrotask](https://developer.mozilla.org/en-US/docs/Web/API/queueMicrotask) is generally a better choice than `nextTick`.

Reply:
Right!
However, it turns out both `queueMicrotask()` and `process.nextTick()` still group both `console.log()` together. But `setTimeout()` does work, so I switched to this instead.